pkg/timescale/write.go

```diff
@@ -112,39 +147,37 @@ func contains(s []string, e string) bool {
 }
 
 func insertRowsTimescale(rowsArray []TimescaleRows, config *config.OutputTimescale) error {
-	var c, parseErr = pgx.ParseConnectionString(config.Connection)
-	if parseErr != nil {
-		return parseErr
-	}
 
-	conn, connErr := pgx.Connect(c)
+	conn, connErr := timescalePools[config.Connection].Acquire(context.Background())
+
 	if connErr != nil {
 		return connErr
 	}
-	defer conn.Close()
 
-	tx, beginErr := conn.Begin()
+	tx, beginErr := conn.Begin(context.Background())
 	if beginErr != nil {
 		return beginErr
 	}
-	defer tx.Rollback() //nolint: errcheck
+	defer tx.Rollback(context.Background()) //nolint: errcheck
 
 	for _, rows := range rowsArray {
-
-		copyCount, err := conn.CopyFrom(pgx.Identifier{rows.TargetTable}, rows.InsertColumns, pgx.CopyFromRows(rows.InsertRows))
-		if err != nil {
-			return fmt.Errorf("Unexpected error for CopyFrom: %v", err)
+		copyCount, copyErr := tx.CopyFrom(context.Background(), []string{rows.TargetTable}, rows.InsertColumns, pgx.CopyFromRows(rows.InsertRows))
+		if copyErr != nil {
+			return fmt.Errorf("Unexpected error for CopyFrom: %v", copyErr)
 		}
+
 		if int(copyCount) != len(rows.InsertRows) {
 			return fmt.Errorf("Expected CopyFrom to return %d copied rows, but got %d", len(rows.InsertRows), copyCount)
 		}
 	}
 
-	err := tx.Commit()
+	err := tx.Commit(context.Background())
 	if err != nil {
 		return err
 	}
 
+	conn.Release()
```

Comment:
That's a bug... if the function returns early because of an error, the connection is never relased... work with `defer` instead?